src/Custom/Chat/ChatClient.cs

```diff
@@ -62,6 +64,7 @@ protected internal ChatClient(ClientPipeline pipeline, string model, Uri endpoin
         _model = model;
         _pipeline = pipeline;
         _endpoint = endpoint;
+        _instrumentation = new InstrumentationFactory(model, endpoint);
```

Comment:
Will we need options in OpenAIClientOptions to configure distributed tracing, e.g. to turn it on or off and or other options like providing an allow-list to prevent secrets from being logged?

Reply:
Yes, great question. 

I believe we'll need addition to base `ClientOptions` to enable/disable/configure tracing/metrics/logging for specific instances of clients. `OpenAIClientOptions` should be able to extend diagnostics options if necessary. 

In this PR tracing/metrics can be enabled globally for all OpenAI clients which is necessary regardless of per-instance configuration/overrides. There are no secrets/sensitive data to redact (yet).

So `ClientOptions` API addition would be purely additive. Do you want me to create issues to follow up?

Reply:
That would be great, and thank you!  @m-redding is working on public APIs for logging options in SCM now - it would be great to make sure we have a unified plan here if there are any overlaps in options for logging and distributed tracing.  @jsquire, @KrzysztofCwalina  FYI.

Reply:
thanks! done - https://github.com/Azure/azure-sdk-for-net/issues/45267